core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -205,7 +205,11 @@ abstract class RawModule extends BaseModule {
       case (true, false) if left.probeInfo.get.writable => ProbeDefine(si, left.lref, RWProbeExpr(Node(right)))
       case (true, false)                                => ProbeDefine(si, left.lref, ProbeExpr(Node(right)))
       case (false, true)                                => Connect(si, left.lref, ProbeRead(Node(right)))
-      case (false, false)                               => Connect(si, left.lref, Node(right))
+      case (false, false) =>
+        (left, right) match {
+          case (_: Property[_], _: Property[_]) => PropAssign(si, left.lref, Node(right))
+          case (_, _) => Connect(si, left.lref, Node(right))
```

Comment:
What should happen for boring from property to data and when/what error will show up? I assume this will blow up later. Is the error message reasonable?

Reply:
Yep Jiuyang had a similar question. I will add an error test and make sure the error is reasonable.

Reply:
Ok, so this logic is only ever used to connect ports/wires created by BoringUtils itself. If the user bores from a Property to a Data, the Property ports will be bored and secret connections from Property to Property made here, but then when the bored output is connected to a non-Property data, you'll get the usual error:

```
Connection between sink (Foo.a: IO[Bool]) and source (chisel3.properties.Property$$anon$9@371acd0e) failed @: Sink (Bool) and Source (chisel3.properties.Property$$anon$9@66435227) have different types.
```